fiber-unix/src/scheduler.ml

```diff
@@ -189,60 +185,19 @@ let time_loop t =
                 active_timer.scheduled +. active_timer.parent.delay
               in
               let need_to_run = scheduled_at < now in
-              if need_to_run then
-                to_run := active_timer :: !to_run
-              else
-                earliest_next :=
-                  Some
-                    (match !earliest_next with
-                    | None -> scheduled_at
-                    | Some v -> min scheduled_at v);
+              if need_to_run then to_run := active_timer :: !to_run;
               not need_to_run));
     let to_run =
       List.sort !to_run ~compare:(fun x y ->
           Timer_id.compare x.parent.timer_id y.parent.timer_id)
       |> List.map ~f:(fun x -> Scheduled x)
     in
     add_events t to_run;
-    Option.iter !earliest_next ~f:(fun s ->
-        with_mutex t.earliest_next_mutex ~f:(fun () ->
-            t.earliest_next <- Some s);
-        match Barrier.signal t.earliest_next_barrier with
-        | Ok () -> ()
-        | Error `Closed -> assert false);
-    with_mutex t.timers_available_mutex ~f:(fun () ->
-        Condition.wait t.timers_available t.timers_available_mutex);
+    Unix.sleepf t.timer_resolution;
```

Comment:
Do I understand correctly that we delegate the scheduling to the OS with the sleep here?